Playground config requires all properties in TypeScript

<!--**Issue Labels**

While not necessary, you can help organize our issues by labeling this issue when you open it.  To add a label automatically, simply [x] mark the appropriate box below:

- [ ] has-reproduction
- [ ] feature
- [ ] docs
- [ ] blocking
- [x ] good first issue

To add a label not listed above, simply place `/label another-label-name` on a line by itself.
-->

ApolloServer constructor accepts parameter `playground?: PlaygroundConfig` where

```typescript
export type PlaygroundConfig =
  | RecursivePartial<PlaygroundRenderPageOptions>
  | boolean;
```

where

```typescript
type RecursivePartial<T> = {
  [P in keyof T]?: T[P] extends (infer U)[]
    ? RecursivePartial<U>[]
    : T[P] extends object ? RecursivePartial<T[P]> : T[P]
};
```

This results in TypeScript requiring one to define all playground options not just the ones that you want to override from defaults.

To reproduce just attempt to create an instance of ApolloServer with playground options not setting all available options:

```typescript
const server = new ApolloServer({
    playground: {
        settings: {
            "request.credentials": "include",
        },
    }
});
```

Failure to provide all options results in:
```
Types of property 'settings' are incompatible.
Argument of type '{ playground: { settings: { "request.credentials": string; }; }; }' is not assignable to parameter of type 'Config'.
  Types of property 'playground' are incompatible.
    Type '{ settings: { "request.credentials": string; }; }' is not assignable to type 'boolean | RecursivePartial<RenderPageOptions> | undefined'.
      Type '{ settings: { "request.credentials": string; }; }' is not assignable to type 'RecursivePartial<RenderPageOptions>'.
        Types of property 'settings' are incompatible.
          Type '{ "request.credentials": string; }' is not assignable to type 'ISettings'.
            Property ''general.betaUpdates'' is missing in type '{ "request.credentials": string; }'.
```

TypeScript version: `3.0.3`.

@JacksonKearl what's the proper way to rebuild the apollo-server-core package after making the above change? I created a tsconfig.json file in node_modules/apollo-core-server and run tsc which gave me a "cannot find module "mock-req" so I just moved that folder and tried to rebuild but I'm getting:
```
➜  apollo-server-core git:(master) ✗ tsc
src/playground.ts:73:9 - error TS2322: Type '{ settings: { 'general.betaUpdates': boolean; 'editor.theme': Theme; 'editor.cursorShape': CursorShape; 'editor.reuseHeaders': boolean; 'tracing.hideTracingResponse': boolean; 'queryPlan.hideQueryPlanResponse': boolean; 'editor.fontSize': number; 'editor.fontFamily': string; 'request.credentials': string; } | { ...;...' is not assignable to type 'RenderPageOptions'.
  Type '{ settings: { 'general.betaUpdates': boolean; 'editor.theme': Theme; 'editor.cursorShape': CursorShape; 'editor.reuseHeaders': boolean; 'tracing.hideTracingResponse': boolean; 'queryPlan.hideQueryPlanResponse': boolean; 'editor.fontSize': number; 'editor.fontFamily': string; 'request.credentials': string; } | { ...;...' is not assignable to type 'RenderPageOptions'.
    Types of property 'schema' are incompatible.
      Type 'RecursivePartial<IntrospectionResult> | undefined' is not assignable to type 'IntrospectionResult | undefined'.
        Type 'RecursivePartial<IntrospectionResult>' is not assignable to type 'IntrospectionResult'.
          Property '__schema' is optional in type 'RecursivePartial<IntrospectionResult>' but required in type 'IntrospectionResult'.

73   const playgroundOptions: PlaygroundRenderPageOptions = {
```
Thanks,
-Steve

--- a/packages/apollo-server-core/src/playground.ts
+++ b/packages/apollo-server-core/src/playground.ts
@@ -16,7 +16,7 @@ const playgroundVersion = '1.7.26';
 type RecursivePartial<T> = {
   [P in keyof T]?: T[P] extends (infer U)[]
     ? RecursivePartial<U>[]
-    : T[P] extends object
+    : T[P] extends (object | undefined)
     ? RecursivePartial<T[P]>
     : T[P];
 };
@@ -63,7 +63,7 @@ export function createPlaygroundOptions(
       }
     : { settings: undefined };
 
-  const playgroundOptions: PlaygroundRenderPageOptions = {
+  const playgroundOptions: any = {
     ...defaultPlaygroundOptions,
     ...playgroundOverrides,
     ...settingsOverrides,

